lib/Lifters/FunctionLifter.cpp

```diff
@@ -1056,43 +1087,47 @@ void FunctionLifter::VisitInstruction(remill::Instruction &inst,
     case remill::Instruction::kCategoryError:
       VisitError(inst, delayed_inst, block);
       break;
-    case remill::Instruction::kCategoryNormal: VisitNormal(inst, block); break;
-    case remill::Instruction::kCategoryNoOp: VisitNoOp(inst, block); break;
+    case remill::Instruction::kCategoryNormal:
+      VisitNormal(inst, block, mapper);
+      break;
+    case remill::Instruction::kCategoryNoOp:
+      VisitNoOp(inst, block, mapper);
+      break;
     case remill::Instruction::kCategoryDirectJump:
-      VisitDirectJump(inst, delayed_inst, block);
+      VisitDirectJump(inst, delayed_inst, block, mapper);
       break;
     case remill::Instruction::kCategoryIndirectJump:
-      VisitIndirectJump(inst, delayed_inst, block);
+      VisitIndirectJump(inst, delayed_inst, block, mapper);
       break;
     case remill::Instruction::kCategoryConditionalIndirectJump:
-      VisitConditionalIndirectJump(inst, delayed_inst, block);
+      VisitConditionalIndirectJump(inst, delayed_inst, block, mapper);
       break;
     case remill::Instruction::kCategoryFunctionReturn:
       VisitFunctionReturn(inst, delayed_inst, block);
```

Comment:
Why does the function return and async hyper call not require the context?

Reply:
Any purely interprocedural control flow does not need the context since no direct instruction lifting happens inside them. ie. for an asynchypercall the function terminates so there is nothing more to decode in this function. The reason we need it for say a condtionalasynchypercall is to handle decoding the fall-through

Reply:
Is it not worth to give all `VisitX` the same prototype?

Reply:
I'm not sure, we'd just be passing an unused parameter then and visits are not used uniformly anyways